### ISSUE TITLE
Midi Binding for Clear Clip

### ISSUE THREAD

**Issue by georgkrause** [openAVproductions/openAV-Luppp#134]
I want to clear a clip with my midi controller.

**Comment by sub26nico** [comment]
+1

**Comment by georgkrause** [comment]
@harryhaaren this commit has a new issue: old midi binding maps wont work anymore. at least the grid launch events does not have any effect anymore.

I checked the version directly before this commit, there it is still working.

**Comment by georgkrause** [comment]
The problem is, that all midi binding maps created before miss that active field. when we release this, we need to provide a tool which can edit the midi maps to work again or find a way make this old maps still working with the new version.

also, the midi learn function need to create valid maps (not the case at the moment)

**Comment by harryhaaren** [comment]
@georgkrause thanks for the latest updated just above: yes good point, the "loading" code should probably error check, and if there's no `active` section, to ignore the clear-clip functionality. I'll add a tag  to this issue to remind that this is a bug. Thanks!

**Comment by georgkrause** [comment]
The problem here is, that in src/controller/binding.hxx:31:

```
public:
	Binding() : status(0), data(0), action(0), active(-1),
		track(-2),scene(-1),send(-1)
	{
		ID = privateID++;
	}
```

The default for active is set to -1. 

To fix this it seems to be as easy as setting the default to 1 if there is no reason against (i tested, works for me)

**Comment by georgkrause** [comment]
Well, this PR fixes the Bug for existing MIDI Maps, but there is still a problem in the bind process of luppp. i tried to bind a button of my apc mini to a clip, its working but the button always clears the clip. I will investigate.

**Comment by georgkrause** [comment]
Should be solved by 817dd63.
